src/Css/CssDocument.php

```diff
@@ -36,11 +37,31 @@ class CssDocument
     private $isImportRuleAllowed = true;
 
     /**
-     * @param string $css
+     * @param string $css {@see https://github.com/squizlabs/PHP_CodeSniffer/issues/3521}
+     * @param bool $debug
+     *        If this is `true`, an exception will be thrown if invalid CSS is encountered.
+     *        Otherwise the parser will try to do the best it can.
      */
-    public function __construct(string $css)
+    public function __construct(string $css, bool $debug)
     {
-        $this->sabberwormCssDocument = (new CssParser($css))->parse();
+        // CSS Parser currently throws exception with nested at-rules (like `@media`) in strict parsing mode
+        $parserSettings = ParserSettings::create()->withLenientParsing(!$debug || static::hasNestedAtRule($css));
+
+        // CSS Parser currently throws exception with non-empty whitespace-only CSS in strict parsing mode, so `trim()`
+        // @see https://github.com/sabberworm/PHP-CSS-Parser/issues/349
+        $this->sabberwormCssDocument = (new CssParser(\trim($css), $parserSettings))->parse();
+    }
+
+    /**
+     * Tests if a string of CSS appears to contain an at-rule with nested rules
+     * (`@media`, `@supports`, `@keyframes`, `@document`,
+     * the latter two additionally with vendor prefixes that may commonly be used).
+     *
+     * @see https://github.com/sabberworm/PHP-CSS-Parser/issues/127
+     */
+    private static function hasNestedAtRule(string $css): bool
```

Comment:
Please let's not make this method static (as there is no need to make it static).

Reply:
OK.  There are arguments either way but have changed.